src/ucm/cuda/replace.c

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "cudamem.h"
+
+#include <ucm/event/event.h>
+#include <ucm/util/log.h>
+#include <ucm/util/reloc.h>
+#include <ucs/sys/compiler.h>
+#include <ucs/sys/preprocessor.h>
+#include <ucs/type/component.h>
+#include <pthread.h>
+
+static pthread_mutex_t ucm_cudamem_get_orig_lock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+static pthread_t volatile ucm_cudamem_get_orig_thread = -1;
+
+
+/**
+ * Define a replacement function to a memory-mapping function call, which calls
+ * the event handler, and if event handler returns error code - calls the original
+ * function.
+ */
+#define UCM_DEFINE_CUDA_FUNC(_name, _rettype, _fail_val, ...) \
```

Comment:
isn't it possible to reuse the existing mmap() macros or extend them if needed?
no copy-paste please..

Reply:
they are almost same, we can extend if we move macros to some common header file.  is that ok?

Reply:
yes

Reply:
moved macros to commons header